src/main/java/org/jenkinsci/plugins/ssegateway/sse/EventDispatcher.java

```diff
@@ -339,11 +339,33 @@ private void dispatchReload() {
             LOGGER.error("Unable to send reload event to client.", e);
         }
     }
+
+    /**
+     * Called on queue events to validate that the disaptcher is still alive
+     */
+    private void validateDispatcher() {
+        Retry retry = retryQueue.peek();
+        if (retry != null) {
+            long ctime = System.currentTimeMillis();
+            long retry_age = (ctime - retry.timestamp);
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("EventDispatcher (%s) - timestamp: %d - current: %d - age: %d", this, retry.timestamp, ctime, retry_age));
+            }
+            // check time of oldest retry envent
+            if (retry_age > RETRY_QUEUE_EVENT_LIFETIME) {
+                // oldest event has timed out - remove all events from retry queue
+                LOGGER.debug("EventDispatcher {} processRetries - clear retryQueue", this);
+                retryQueue.clear();
+            }
+        }
+        checkDispatcherFailTimeout("dispatcher.validation");
```

Comment:
The other code was copy-pasted, but this is new. IIUC, this checks if the last success was too old, and if so, clears the queue and removes all subscribers for this dispatcher. Do we need to do something so that if this method is called from `addToRetryQueue`, that method stops trying to send the message? Otherwise it will immediately restart the queue, which is pointless because we don't have subscribers anymore, right? Maybe the condition on line 370 should also also be true if the set of subscribers is empty?

Reply:
Sounds reasonable, I added it. I'm not a fan of this code but the constant memory leaks over a month can really bog down an instance.